### PR TITLE
Osd 4275 Metrics and Monitoring for the Shredder 

### PR DESCRIPTION
**Pull request by amolgautam25**
Metrics and Monitoring for the Shredder :
Shredder exposes default prometheus metrics over a service

### CODE REVIEW

File: deploy/service_account_role.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  namespace: aws-account-shredder
+  name: read-access
```
Comment (fahlmant) [inline deploy/service_account_role.yaml:5]: Can we change this to just `aws-account-shredder`?

File: main.go [hunk 2]
```diff
@@ -32,10 +37,31 @@ func main() {
...
+
+	//Create localMetrics endpoint and register localMetrics
+	metricsServer := metrics.NewBuilder().WithPort(metricsPort).WithPath(metricsPath).
+		WithCollectors(localMetrics.MetricsList).
+		WithRoute().
+		WithServiceName("aws-account-shredder").
+		GetConfig()
+
+	// Configure localMetrics if it errors log the error but continue
+	// isa this required ????
```
Comment (fahlmant) [inline main.go:59]: Yes this is required.

File: pkg/localMetrics/localMetrics.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
+package localMetrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	MetricTotalAWSAccounts = prometheus.NewGaugeVec(prometheus.GaugeOpts{
```
Comment (fahlmant) [inline pkg/localMetrics/localMetrics.go:8]: Please only have one dummy metric here. You don't need all of these.

Comment (amolgautam25) [reply pkg/localMetrics/localMetrics.go:8]: done

File: deploy/deployment.yaml [hunk 4]
```diff
@@ -3,18 +3,21 @@ kind: Deployment
...
       labels:
-        app: aws-account-shredder
+        name: aws-account-shredder
     spec:
       serviceAccountName: aws-account-shredder
       containers:
-      - name: aws-account-operator
-        image: quay.io/agautam/aws-account-shredder:latest
+        - name: aws-account-shredder
+          image: quay.io/agautam/aws-account-shredder:test1
```
Comment (fahlmant) [inline deploy/deployment.yaml:20]: Please put this back to `:latest`

File: pkg/localMetrics/localMetrics.go [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
```
Comment (fahlmant) [inline pkg/localMetrics/localMetrics.go:8]: I think you can just use `NewGauge` here. Can you also rename this metric to DummyMetric?

**Pull request merged**